js/ai/src/tool.ts

```diff
@@ -381,3 +390,36 @@ function interruptTool(registry: Registry) {
     throw new ToolInterruptError(metadata);
   };
 }
+
+/**
+ * Defines a dynamic tool. Dynamic tools are just like regular tools but will not be registered in the
+ * Genkit registry and can be defined dynamically at runtime.
+ */
+export function dynamicTool<I extends z.ZodTypeAny, O extends z.ZodTypeAny>(
```

Comment:
If the call signature doesn't include the interrupt version, will it work with a non-beta registry? If so, could we maybe allow it to work with stable and use assertUnstable only if it's an interrupt?

Reply:
yes, beta is only asserted if interrupt is involved. It will work with `fn` in stable.